backend/server/routers/planner.py

```diff
@@ -91,11 +88,24 @@ def validate_term_planner(plannerData: PlannerData):
         }
     }
 )
-def autoplanning(courseCodes: List, userData: UserData, programTime: ProgramTime) -> Dict:
-    user = User(dict(userData))
-
+def autoplanning(courseCodes: List, plannerData: PlannerData, programTime: ProgramTime) -> Dict:
+    user = extract_user_from_planner_data(plannerData)
+    autoplanned = []
     try:
+        # Convert list of course codes to list of Course objects 
         courses = [get_course_object(courseCode, programTime) for courseCode in courseCodes]
+
+        # Convert plannerData to list of Course object and FORCE the term the course is taken in
+        # e.g. if the user has taken COMP1511 in 2022 T1, then the autoplanning will have COMP1511 in 2022 T1 
+        # regardless of the actual COMP1511 term offerings
```

Comment:
wait actually i forgot about this problem... i think my solution would have forced a course placed in the wrong term to always be infeasable 😅

Reply:
ooo what's your solution? 👀

Reply:
basically figure out what the index the user set that would correspond to the number in the model, then hardcode it
```python
i = course.getIndexPlaced() # assume this produces like 0 if they placed it in summer term first year or 4 if its t3 first year
model.Add(course.getVariable() == i)
```
... then let the rest of the algo run normally. This lets the model see that the variable is there, but doesnt give it the flexibility to move it and is instead forced to work around it.

Reply:
looks neat, let me steal that real quick xD

jks i think we should keep the current one since it allows past terms to be weird, also i'm not a fan of having routers data model imported to algorithms code and we need that O(n^3) conversion loop to get courses marks anyway. Oops forgot to do so xD